### PR TITLE
Documentation: Fix typos

### PR DESCRIPTION
**Pull request by dantleech**
This PR contains grammer fixes.

### CODE REVIEW

File: Documentation/En/Index.xyl [hunk 1]
```diff
@@ -26,28 +26,28 @@
...
   part is: <q>“what <em>we</em> expect”</q>. By extension, we understand a
-  test does not check the SUT is <strong>verified</strong>.</p>
+  test does not check that the SUT is <strong>verified</strong>.</p>
   <blockquote cite="Dijkstra">Testing shows the presence, not the absence of
   bugs.</blockquote>
   <p>A test only checks the SUT is <strong>valid</strong>. The difference is
   that a valid system has no bug for some <strong>non exhaustive
   executions</strong>, but a verified system does what it is designed to do.
-  Despite tests do not proof the absence of bugs, they are useful (easy to
+  Despite this, tests do not proove the absence of bugs, they are useful (easy to
```
Comment (dantleech) [inline Documentation/En/Index.xyl]: s/proove/prove

File: Documentation/En/Index.xyl [hunk 2]
```diff
@@ -77,25 +77,25 @@
   <p>The “Conception support” axis describes the <strong>visibility</strong> we
   have on the System Under Test: Either <strong>black box</strong> or
   <strong>white box</strong>. Their meanings are different according to the
-  context but we can consider a system as a white box if we look and use what
-  is <strong>inside</strong> the box, else this is a black box. For instance,
-  for unit tests, let say with a function as the SUT, black box testing will
+  context but we can consider a system as a white box if we look at and use what
+  is <strong>inside</strong> the box, else it is a black box. For instance,
+  with unit tests, lets say with a function as the SUT, black box testing will
```
Comment (dantleech) [inline Documentation/En/Index.xyl]: s/lets/let's ?

File: Documentation/En/Index.xyl [hunk 3]
```diff
@@ -103,9 +103,9 @@
   <strong>unit</strong> and <strong>functional</strong> tests. We also include
   performance tests in some libraries.</p>
   <p>Remember that each point in this test universe may imply different tools
-  and different development practices. When talking about tests, this is
-  important to have this schema in minds and to wonder where our problematic
-  locates.</p>
+  and different development practices. When talking about tests, it is
+  important to have this schema in minds and to consider where our problematic
```
Comment (dantleech) [inline Documentation/En/Index.xyl]: s/minds/mind

File: Documentation/En/Index.xyl [hunk 4]
```diff
@@ -279,8 +279,8 @@ class Bar extends Test\Unit\Suite
     // Then:
     $this->integer($result)>-isEqualTo(3);
 }</code></pre>
-  <p>We however use the first form, which is clearer: Indentation, special
-  keywords (through method names) and force us to clearly separate these
+  <p>We use, however, the first form. It is clearer: Indentation, special
+  keywords (through method names) and forces us to clearly separate these
```
Comment (dantleech) [inline Documentation/En/Index.xyl]: s/and /and it

File: Documentation/En/Index.xyl [hunk 5]
```diff
@@ -14,7 +14,7 @@
 
   <h2 id="Introduction" for="main-toc">Introduction</h2>
 
-  <p>Software quality is a vast topic in Computer Science. There is a lot of
+  <p>Software quality is a vast topic in Computer Science. There are a lot of
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: Hmm, nop, it's “there is” here :-].

File: Documentation/En/Index.xyl [hunk 6]
```diff
@@ -26,28 +26,28 @@
   abbreviated SUT, and is composed of two parts:</p>
   <ol>
     <li><strong>Test data</strong> to execute the SUT, and</li>
-    <li>An <strong>oracle</strong>, establishing the <strong>test
+    <li>an <strong>oracle</strong>, establishing the <strong>test
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: In English, we put the first word of every item of a list with an upper-case. This is not the case in other languages like French or Italian, or even Arabic though.

File: Documentation/En/Index.xyl [hunk 7]
```diff
@@ -26,28 +26,28 @@
...
     verdict</strong>: Are the result of the execution and the state of
-    the SUT after its execution the ones we expect or not?</li>
+    the SUT after it's execution the ones we expect or not?</li>
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: No, it's “its”. Possessive.

File: Documentation/En/Index.xyl [hunk 8]
```diff
@@ -26,28 +26,28 @@
...
   </ol>
   <p>The values of the verdict are: <strong>Success</strong>,
   <strong>fail</strong> or <strong>inconclusive</strong>. A success means this
   is what we expect, a fail means this is not what we expect and an
-  inconclusive means this is not possible to give an answer. The important
+  inconclusive means it is not possible to give an answer. The important
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: Why “it” and not “this” here?

File: Documentation/En/Index.xyl [hunk 9]
```diff
@@ -26,28 +26,28 @@
...
   part is: <q>“what <em>we</em> expect”</q>. By extension, we understand a
-  test does not check the SUT is <strong>verified</strong>.</p>
+  test does not check that the SUT is <strong>verified</strong>.</p>
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: “that” is optional here. We can skip it.

File: Documentation/En/Index.xyl (hunk 1 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: “this” is not necessary here, isn't it?

Comment (Hywan) [inline Documentation/En/Index.xyl]: :+1: for
```diff
- proof
+ prove
```

File: Documentation/En/Index.xyl [hunk 10]
```diff
@@ -26,28 +26,28 @@
...
   <blockquote cite="Dijkstra">Testing shows the presence, not the absence of
   bugs.</blockquote>
   <p>A test only checks the SUT is <strong>valid</strong>. The difference is
   that a valid system has no bug for some <strong>non exhaustive
   executions</strong>, but a verified system does what it is designed to do.
-  Despite tests do not proof the absence of bugs, they are useful (easy to
+  Despite this, tests do not proove the absence of bugs, they are useful (easy to
   write, easy to read, not all system can be proven etc.). Nevertheless,
-  depending of the chosen test data, tests can be close to proofs. As an
+  depending on the chosen test data, tests can be close to proofs. As an
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: :+1:

File: Documentation/En/Index.xyl [hunk 11]
```diff
@@ -26,28 +26,28 @@
...
   example, a test with exhaustive test data can be considered as a proof.</p>
   <p>A test can be written <strong>manually</strong>, for instance by a test
   engineer. A SUT is provided, the engineer uses test data to execute it and
-  the test verdict is established by the engineer itself. In the case of an
+  the test verdict is established by the engineer. In the case of an
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: Why removing “itself”?

File: Documentation/En/Index.xyl [hunk 12]
```diff
@@ -56,7 +56,7 @@
   <strong>automatic</strong> tests are generated and executed by the machine.
   The machine may rely on a specification, like a contract, to automatically
   generate the tests (please, see <a href="@hack:chapter=Praspel">the
-  <code>Hoa\Praspel</code> library</a> or the research papers bellow we have
+  <code>Hoa\Praspel</code> library</a> or the research papers below that we have
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: :+1:

File: Documentation/En/Index.xyl (hunk 2 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: Yup “let's”. Could you add this to your patch please?

File: Documentation/En/Index.xyl [hunk 13]
```diff
@@ -77,25 +77,25 @@
   <p>The “Conception support” axis describes the <strong>visibility</strong> we
   have on the System Under Test: Either <strong>black box</strong> or
   <strong>white box</strong>. Their meanings are different according to the
-  context but we can consider a system as a white box if we look and use what
-  is <strong>inside</strong> the box, else this is a black box. For instance,
-  for unit tests, let say with a function as the SUT, black box testing will
+  context but we can consider a system as a white box if we look at and use what
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: :+1:

File: Documentation/En/Index.xyl [hunk 14]
```diff
@@ -77,25 +77,25 @@
...
+  is <strong>inside</strong> the box, else it is a black box. For instance,
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: :+1:

File: Documentation/En/Index.xyl (hunk 2 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: :+1: for “with”.

File: Documentation/En/Index.xyl [hunk 15]
```diff
@@ -77,25 +77,25 @@
...
-  <strong>integration</strong> of these components inside other ones, which
-  together form the system. All levels are important: This is not because a
-  function does what it is supposed to do that it will integrate nicely with
-  another one.</p>
+  altitude) the system will be tested: From the lowest one,
+  <strong>unit</strong> (a function or a method) to the highest one, the whole
+  <strong>system</strong>, which includes <strong>components</strong> (a set of
+  functions) and the <strong>integration</strong> of these components inside
+  other ones, which together form the system. All levels are important: This is
+  not because a function does what it is supposed to do, but that it will integrate
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: Why “but”? It changes the meaning of the sentence here.

File: Documentation/En/Index.xyl [hunk 16]
```diff
@@ -77,25 +77,25 @@
...
+  nicely with another one.</p>
   <p>Finally, the “Type of test” axis describes the <strong>goal</strong> of the
   tests: Either <strong>functional</strong> (“everything works as expected and
-  that's it”), <strong>robustness</strong> (“this is not possible to put the
+  that's it”), <strong>robustness</strong> (“it is not possible to put the
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: :+1:

File: Documentation/En/Index.xyl [hunk 17]
```diff
@@ -103,9 +103,9 @@
   <strong>unit</strong> and <strong>functional</strong> tests. We also include
   performance tests in some libraries.</p>
   <p>Remember that each point in this test universe may imply different tools
-  and different development practices. When talking about tests, this is
-  important to have this schema in minds and to wonder where our problematic
-  locates.</p>
+  and different development practices. When talking about tests, it is
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: Why “it” here?

File: Documentation/En/Index.xyl (hunk 3 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: :+1:

File: Documentation/En/Index.xyl [hunk 18]
```diff
@@ -114,13 +114,13 @@
   data) that fixes the values of the arguments (also called inputs) of the
   SUT. A <strong>test case</strong> is a pair composed of a state (the context
   in which the test is executed) and a test. A <strong>test suite</strong> is
-  a set of test cases. Therefore, there is two states from the test point of
+  a set of test cases. Therefore, there are two states from the test point of
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: “there is” no?

File: Documentation/En/Index.xyl [hunk 19]
```diff
@@ -114,13 +114,13 @@
...
   view: A <strong>pre-state</strong> —before the execution of the SUT— and a
   <strong>post-state</strong> —after the execution—.</p>
   <p>Building the pre-state is something crucial: The goal is to put the SUT into a
   specific (pre-)state in order to test it. The part of the test case
-  responsible to create the pre-state is called the <strong>preamble</strong>.
-  Thus, the preambler is a code that puts the SUT into a specific state.</p>
+  responsible for creating the pre-state is called the <strong>preamble</strong>.
+  Thus, the preambler is code that puts the SUT into a specific state.</p>
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: Why removing the singularity form here?

File: Documentation/En/Index.xyl [hunk 20]
```diff
@@ -114,13 +114,13 @@
...
   in which the test is executed) and a test. A <strong>test suite</strong> is
-  a set of test cases. Therefore, there is two states from the test point of
+  a set of test cases. Therefore, there are two states from the test point of
   view: A <strong>pre-state</strong> —before the execution of the SUT— and a
   <strong>post-state</strong> —after the execution—.</p>
   <p>Building the pre-state is something crucial: The goal is to put the SUT into a
   specific (pre-)state in order to test it. The part of the test case
-  responsible to create the pre-state is called the <strong>preamble</strong>.
-  Thus, the preambler is a code that puts the SUT into a specific state.</p>
+  responsible for creating the pre-state is called the <strong>preamble</strong>.
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: Actually, this is not something clear for me! Why using “for +ing” instead of “to +infinitive” here? Do you know the rule?

File: Documentation/En/Index.xyl [hunk 21]
```diff
@@ -129,23 +129,23 @@
     <li>The <strong>oracle</strong>, to compute the test verdict (based on the
     execution result and the post-state).</li>
   </ol>
-  <p>The oracle consists of a sequence of assertions: “Is this value equals
+  <p>The oracle consists of a sequence of assertions: “Is this value equal
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: :+1:

File: Documentation/En/Index.xyl [hunk 22]
```diff
@@ -129,23 +129,23 @@
...
   to…?”, “Is this value changed this way…?” etc. It is supposed to be able to
   check the form of some data as much as comparing other data between the
   pre-state and the post-state.</p>
   <p>In a test case, we represent and introduce the preamble, the execution and
   the oracle respectively by the <code>given</code>, <code>when</code> and
-  <code>then</code> keywords. This will be detail hereinafter.</p>
+  <code>then</code> keywords. This will be detailed hereinafter.</p>
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: :+1:

File: Documentation/En/Index.xyl [hunk 23]
```diff
@@ -129,23 +129,23 @@
...
 
   <h3 id="Test_framework" for="main-toc">Test framework</h3>
 
-  <p>Hoa did not developed its own test framework for several reasons. The main
+  <p>Hoa did not develop its own test framework for several reasons. The main
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: :+1:

File: Documentation/En/Index.xyl [hunk 24]
```diff
@@ -129,23 +129,23 @@
...
   one is that this is a laborious task. It implies more development, more tools
-  to maintain, less people focused on Hoa etc. Instead, we prefer to choose a
-  good project to rely on and contribute to it. Then, we choose
+  to maintain, less people focused on Hoa etc. Instead, we chose to rely on a
+  good project and contribute to it. We chose
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: :+1:

File: Documentation/En/Index.xyl [hunk 25]
```diff
@@ -227,9 +227,9 @@ class Bar extends Test\Unit\Suite
   it needs to be build. In this case, <code>LUT</code> could be defined to
   <code>Hoa\Foo</code>.</p>
   <p>The <code>Bar</code> class extends the <code>Hoa\Test\Unit\Suite</code>
-  class, which defines a unit test suite. It offers all the test API we need
+  class, which defines a unit test suite. It offers the complete test API needed
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: :+1:

File: Documentation/En/Index.xyl [hunk 26]
```diff
@@ -227,9 +227,9 @@ class Bar extends Test\Unit\Suite
...
   to write tests.</p>
-  <p>Test cases are public method. They do not use a camel case notation but an
+  <p>Test cases are public methods. They do not use a camel case notation but an
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: :+1:

File: Documentation/En/Index.xyl (hunk 4 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: :+1:

File: Documentation/En/Index.xyl [hunk 27]
```diff
@@ -505,7 +505,7 @@ class Bar extends Test\Unit\Suite
   <pre><code class="language-shell">$ vendor/bin/hoa test:run --directories Test/ --debug</code></pre>
   <p>The order of the option does not matter, as usual with the <code>hoa</code>
   command.</p>
-  <p>To produce a debugging information, we should use the <code>dump</code>
+  <p>To produce debugging information, we should use the <code>dump</code>
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: :+1:

File: Documentation/En/Index.xyl [hunk 28]
```diff
@@ -526,7 +526,7 @@ class Bar extends Test\Unit\Suite
   <h3 id="Select_tests_to_run" for="main-toc">Select tests to run</h3>
 
   <p>Selecting tests to run enables a <strong>faster feedback</strong> and a
-  <strong>shorter “test loop”</strong>. We previously saw the
+  <strong>shorter “test loop”</strong>. We previously used the
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: :+1:

File: Documentation/En/Index.xyl [hunk 29]
```diff
@@ -583,22 +583,23 @@ public function case_sum()
   <h3 id="Choose_the_PHP_virtual_machine" for="main-toc">Choose the PHP virtual
   machine</h3>
 
-  <p>When testing, this is also useful to be able to select one specific
+  <p>When testing, it is also useful to be able to select one specific
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: :+1:

File: Documentation/En/Index.xyl [hunk 30]
```diff
@@ -583,22 +583,23 @@ public function case_sum()
...
   <strong>PHP virtual machine</strong>, like <a href="http://php.net/">a
   specific version of PHP</a> (the default virtual machine) or
   <a href="http://hhvm.com/">the latest version of HHVM</a> for instance. There
-  is several PHP virtual machines nowadays and we cannot ignore them.</p>
+  are several PHP virtual machines nowadays and we cannot ignore them.</p>
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: Again, why “there are”?

File: Documentation/En/Index.xyl [hunk 31]
```diff
@@ -583,22 +583,23 @@ public function case_sum()
...
-  <p>We can imagine the <code>--php-binary</code> option value defined by a
-  global variable. Thus, we could re-use the same script to execute tests
-  against several PHP virtual machines, like:</p>
-  <pre><code class="language-shell">$ PHP_BIN=/usr/bin/php vendor/bin/hoa test:run --directories … --php-binary $PHP_BIN</code></pre>
-  <p>Please, remember to use PHP syntax and features described in
-  <a href="https://github.com/php/php-langspec">the PHP specification</a> as
-  most as possible.</p>
+  <p>We can imagine that the <code>--php-binary</code> option value is determined by a
+  global variable. In this case we could re-use the same script to execute
+  tests against several PHP virtual machines, like:</p> <pre><code
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: Bad re-formatting here.

File: Documentation/En/Index.xyl [hunk 32]
```diff
@@ -583,22 +583,23 @@ public function case_sum()
...
   binary; thus, for instance:</p>
   <pre><code class="language-shell">$ vendor/bin/hoa test:run --directories … --php-binary /usr/bin/php</code></pre>
-  <p>We can imagine the <code>--php-binary</code> option value defined by a
-  global variable. Thus, we could re-use the same script to execute tests
-  against several PHP virtual machines, like:</p>
-  <pre><code class="language-shell">$ PHP_BIN=/usr/bin/php vendor/bin/hoa test:run --directories … --php-binary $PHP_BIN</code></pre>
-  <p>Please, remember to use PHP syntax and features described in
-  <a href="https://github.com/php/php-langspec">the PHP specification</a> as
-  most as possible.</p>
+  <p>We can imagine that the <code>--php-binary</code> option value is determined by a
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: “that” is optional but it's fine.
:+1:

File: Documentation/En/Index.xyl [hunk 33]
```diff
@@ -611,20 +612,20 @@ public function case_sum()
     <li><strong>Concurrent</strong>, “all” test cases at the same time in
     separated processes.</li>
   </ul>
-  <p>When running tests, isolation is really important: No memory conflict, no
-  execution conflict, state of the System Under Test is reset each time etc.
+  <p>When running tests, isolation is really important: No memory conflicts, no
+  execution conflicts, the state of the System Under Test is reset each time etc.
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: Why plural form if there is a quantity of 0 :-p (“no” = 0)?

File: Documentation/En/Index.xyl [hunk 34]
```diff
@@ -611,20 +612,20 @@ public function case_sum()
...
   The test verdict does not depend of a previous run and is therefore
-  <strong>deterministic</strong> and <strong>unambiguous</strong> (also if the
-  SUT is).</p>
+  <strong>deterministic</strong> and <strong>unambiguous</strong> (in as far as
+  the SUT is).</p>
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: :+1:

File: Documentation/En/Index.xyl [hunk 35]
```diff
@@ -611,20 +612,20 @@ public function case_sum()
...
   <p>By default, <code>Hoa\Test</code> will use the <strong>concurrent</strong>
   test execution engine. All test cases are not executed at the same time; in
   fact the number of test cases to run is defined by the number of processes to
   use. By default, this is either 2 or 4 depending of the platform but we can
   specify it by using the <code>--concurrent-processes</code> option. Thus, to
-  force using 1024 processes to run tests (assuming we have a monster
-  computer):</p>
+  force the use of 1024 processes when running the tests (assuming that we have
+  a monster computer):</p>
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: :+1:

File: Documentation/En/Index.xyl [hunk 36]
```diff
@@ -611,20 +612,20 @@ public function case_sum()
...
   <pre><code class="language-shell">$ vendor/bin/hoa test:run --directories Test/ --concurrent-processes 1024</code></pre>
-  <p>Because the inline test execution engine is not interesting for Hoa's
+  <p>Because the inline test execution engine is not relevant for Hoa's
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: :+1:

File: Documentation/En/Index.xyl [hunk 37]
```diff
@@ -640,7 +641,7 @@ public function case_sum()
   to a <strong>virtual file</strong> or a <strong>directory</strong>.
   Additional <strong>query strings</strong> can be present to specify more
   information, such as permissions, access time etc. The parent-child relation
-  is automatically created, not need to specify that a file is a child of a
+  is automatically created, it is not necessary to specify that a file is a child of a
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: Actually the former idea was:
```diff
- not need
+ no need
```
Would it be correct?

**Comment by Hywan** [comment]
Thank you very much for this deep corrections! I am ashamed there is so much corrections to do :-(. Well, there is some open questions. Keep going!

File: Documentation/En/Index.xyl (hunk 1 above)
Comment (dantleech) [inline Documentation/En/Index.xyl]: I think `this` is required, otherwise it reads as if `despite` were a type of test. However I think despite could be replaced with `nevertheless` (néanmoins?)

File: Documentation/En/Index.xyl (hunk 5 above)
Comment (dantleech) [inline Documentation/En/Index.xyl]: I think `are` should be used for plural forms, e.g. `there is a lot of rain`, `there are a lot of penguins`.

File: Documentation/En/Index.xyl (hunk 6 above)
Comment (dantleech) [inline Documentation/En/Index.xyl]: ok, I thought this was an inline list for some reason, will change back.

File: Documentation/En/Index.xyl (hunk 7 above)
Comment (dantleech) [inline Documentation/En/Index.xyl]: `it's` is possesive, `its` is plural

File: Documentation/En/Index.xyl (hunk 1 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: No I guess it's fine here. This is a form I met often in the literature.

File: Documentation/En/Index.xyl (hunk 8 above)
Comment (dantleech) [inline Documentation/En/Index.xyl]: I think because `this` is specifc, it refers to something. I would rewrord this as `... means that it is ...` or maybe rewrod the paragraph something like the following:

````
"Success" means that the results is what we expected, "fail" means that the result was not as expected and
inconclusive means that it was not possible to determine if the result was either a success or failure.
````

File: Documentation/En/Index.xyl (hunk 5 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: No, that's more subtle. http://www.grammar.cl/Present/ThereIsThereAre.htm
“lot of strategies”, this is not countable, so “there is” is fine.

File: Documentation/En/Index.xyl (hunk 6 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: Even for an line list: It applies!

File: Documentation/En/Index.xyl (hunk 7 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: its (singular) = their (plural).

File: Documentation/En/Index.xyl (hunk 9 above)
Comment (dantleech) [inline Documentation/En/Index.xyl]: For me it reads more naturally with `that`, but I can drop it.

File: Documentation/En/Index.xyl (hunk 11 above)
Comment (dantleech) [inline Documentation/En/Index.xyl]: You can't refer to a person as `it`, we could say `himself` but that's not politically correct, I think it's just easier to drop it.

File: Documentation/En/Index.xyl (hunk 8 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: I am afraid this is the opposite: “it” is specific and “this” is not specific :-/.

File: Documentation/En/Index.xyl (hunk 9 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: Please yes :-).

File: Documentation/En/Index.xyl (hunk 11 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: Correct! Sorry :-).

Comment (Hywan) [inline Documentation/En/Index.xyl]: So :+1:

File: Documentation/En/Index.xyl (hunk 15 above)
Comment (dantleech) [inline Documentation/En/Index.xyl]: Actually, I am not sure of the meaning of this sentance. 

`that is will integrate nicely with another one` applies only to functional/system tests? `function does what it is supposed to do` applies to unit tests.

Could you reword it perhaps?

File: Documentation/En/Index.xyl (hunk 17 above)
Comment (dantleech) [inline Documentation/En/Index.xyl]: It's the same in French no? `quand on parle de tests, c'est important ...`

File: Documentation/En/Index.xyl (hunk 18 above)
Comment (dantleech) [inline Documentation/En/Index.xyl]: there are = plural

File: Documentation/En/Index.xyl (hunk 20 above)
Comment (dantleech) [inline Documentation/En/Index.xyl]: I am a native speaker, so I don't know much about the language theory, but, here the test case is responsible *for* something. `to` implies "direction" here, `he is responsible to the director` (he must answer to the director), is different from `he is responsible for the director` (he takes care of the director).

File: Documentation/En/Index.xyl (hunk 15 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: Actually the first formulation is correct. What is your language? I can help by translating to your first language?

File: Documentation/En/Index.xyl (hunk 17 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: Sure, but “this is” is not specific, while “it is” is specific. But nevermind, in this particular case it works.

So :+1:.

File: Documentation/En/Index.xyl (hunk 18 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: :+1: correct, “two states” are countable.

File: Documentation/En/Index.xyl (hunk 19 above)
Comment (dantleech) [inline Documentation/En/Index.xyl]: we are taling about one or multiple lines of code? `a code` doesn't sound right, but `some code` would be better.

File: Documentation/En/Index.xyl (hunk 20 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: Ok, thanks!

So :+1:.

File: Documentation/En/Index.xyl (hunk 30 above)
Comment (dantleech) [inline Documentation/En/Index.xyl]: plural

File: Documentation/En/Index.xyl (hunk 19 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: Ok, thanks for the explanation!

Comment (Hywan) [inline Documentation/En/Index.xyl]: Or maybe “a piece of code”?

Whatever, it's a :+1: here.

File: Documentation/En/Index.xyl (hunk 30 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: It's not countable, so “there is”, nop?

File: Documentation/En/Index.xyl (hunk 33 above)
Comment (dantleech) [inline Documentation/En/Index.xyl]: Because there could be many classes of memory or execution conflicts. We are not talking about a specific instance of a memory conflict, but memory conflicts in general. I would write it as I have edited it, but whatever makes more sense to you.

File: Documentation/En/Index.xyl (hunk 37 above)
Comment (dantleech) [inline Documentation/En/Index.xyl]: in that case `there is no need to specify ...` but `not necessary` hold the same meaning.

File: Documentation/En/Index.xyl (hunk 5 above)
Comment (dantleech) [inline Documentation/En/Index.xyl]: Hmm, yeah if `lot` is the subject, then `is` is OK. But as an English reader this sounds strange and I have to read it and emphasise "lot". For me it would be less ambiguous as:

````
There are many strategies
````

File: Documentation/En/Index.xyl (hunk 7 above)
Comment (dantleech) [inline Documentation/En/Index.xyl]: Your right: https://en.wiktionary.org/wiki/its#English :)

File: Documentation/En/Index.xyl (hunk 8 above)
Comment (dantleech) [inline Documentation/En/Index.xyl]: Sorry, I didn't explain correctly. Basically the sentance does not read correctly to me in its current form. I would change it to `means that it is not ...`. Sorry I can't explain better.

File: Documentation/En/Index.xyl (hunk 15 above)
Comment (dantleech) [inline Documentation/En/Index.xyl]: English :) The sentance seems to imply that `all levels are important to ensure that a function integrates with other functions in the system` - but this is the role of the integration testing level. Unit testing is responsible for ensuring that a functional does what it is supposed to do.

Could you rephrase it so that I could understand better? (I might just be reading it incorrectly).

File: Documentation/En/Index.xyl (hunk 30 above)
Comment (dantleech) [inline Documentation/En/Index.xyl]: Not sure tbh, but `is` sounds wrong - maybe `there exist` ?

File: Documentation/En/Index.xyl (hunk 1 above)
Comment (dantleech) [inline Documentation/En/Index.xyl]: So we keep `despite this` or .. ?

**Comment by dantleech** [comment]
Updated..

File: Documentation/En/Index.xyl (hunk 33 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: It's fine for me!
:+1:

File: Documentation/En/Index.xyl (hunk 37 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: :+1:

File: Documentation/En/Index.xyl (hunk 5 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: Please, apply your last proposal :-).

File: Documentation/En/Index.xyl (hunk 8 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: Correct. Please, apply your proposal :-).

File: Documentation/En/Index.xyl (hunk 15 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: https://github.com/hoaproject/Test/pull/49/files#r42718023 is correct.
Unit testing = a function does what it is supposed to do.
Functional system = functions integrate nicely between each others.

File: Documentation/En/Index.xyl (hunk 30 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: Yup, +1 for “there exist”. Please, apply this proposal :-).

File: Documentation/En/Index.xyl (hunk 1 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: :+1: yup

**Comment by Hywan** [comment]
Thank you for your effort!

File: Documentation/En/Index.xyl (hunk 15 above)
Comment (dantleech) [inline Documentation/En/Index.xyl]: ok, I think I read it correctly now - if I substitute "This is not because" with "It is not because" then it makes sense somehow).

````
All levels are important: It is not that a function does what it is supposed to do,
but that it integrates correctly with others.
````

But even then, it seems to suggest that Unit testing is less important. I would say that Unit Testing is the most important level, as in addition to allowing you to test all the code paths in a Unit, it also informs good design. Having only functional tests in a library ensures that the system works, but ignores its architecture. For me it is important *both* that a Unit does what it is supposed to do and that it integrates well with others.

**Comment by dantleech** [comment]
No problem, its a pleasure to contribute :) I have fixed the things we agreed on and there is only the "All levels are important" paragraph to resolve.

File: Documentation/En/Index.xyl (hunk 15 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: Yup, they are both very important, but they do no test the same thing.

Maybe:
> All levels are important: While unit tests ensure a function does what it is supposed to, they do not ensure a function integrates nicely wit hother ones in the system. So, levels are complementary.

Thoughts?

**Comment by Hywan** [comment]
Perfect!

**Comment by dantleech** [comment]
I think what you proposed for "All levels are important" makes good sense, have updated.

**Comment by Hywan** [comment]
So everything is ready for a new review?

**Comment by Hywan** [comment]
Get ready for a squash to get one commit, named: “Documentation: Clarify and fix several typos” (or something equivalent).

**Comment by dantleech** [comment]
Yep, ready for final review. Have squashed.

File: Documentation/En/Index.xyl [hunk 38]
```diff
@@ -31,9 +31,10 @@
     the SUT after its execution the ones we expect or not?</li>
   </ol>
   <p>The values of the verdict are: <strong>Success</strong>,
-  <strong>fail</strong> or <strong>inconclusive</strong>. A success means this
-  is what we expect, a fail means this is not what we expect and an
-  inconclusive means this is not possible to give an answer. The important
+  <strong>fail</strong> or <strong>inconclusive</strong>. "Success" means that
```
Comment (Hywan) [inline Documentation/En/Index.xyl]: Please, use `“Success”` (real English quotes) instead of `"Success"` (straight coding quotes).

Comment (Hywan) [inline Documentation/En/Index.xyl]: `“` = http://graphemica.com/%E2%80%9C
`”`= http://graphemica.com/%E2%80%9D

File: Documentation/En/Index.xyl [hunk 39]
```diff
@@ -31,9 +31,10 @@
...
+  the results is what we expected, "fail" means that the result was not as
```
Comment (Hywan) [inline Documentation/En/Index.xyl]:
```diff
- the results
+ the result
```
There is one single result.

Comment (Hywan) [inline Documentation/En/Index.xyl]:
```diff
- "fail"
+ “fail”
```

**Comment by Hywan** [comment]
3 tiny comments and we are good!

**Comment by dantleech** [comment]
Amended the commit, everything good from here.

**Comment by Hywan** [comment]
Thank you very much!

**Comment by dantleech** [comment]
Thankyou, glad to contribute to a great project :)

**Comment by Hywan** [comment]
Welcome on board: http://hoa-project.net/Community.html#Contributors :wink:.

**Comment by dantleech** [comment]
Thanks!
